### PR TITLE
Add contract verification to ERC-20 tutorial

### PR DESCRIPTION
**Pull request by trruckerfling**

### CODE REVIEW

File: content/en/smart-contracts/fundamentals/erc-20-quickstart/index.md [hunk 1]
```diff
@@ -213,6 +213,34 @@ Now that we've successfully compiled our contract, we need to deploy it somewher
 
 On the Filecoin network, a new set of blocks, also called a tipset, is created every thirty seconds. When deploying a contract, the transaction needs to be received by the network, and then the network needs to confirm the contract. This process takes around one to two tipsets to process -- or around 60 to 90 seconds.
 
+## Verify your contract (main net deployment only)
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/erc-20-quickstart/index.md:216]: Note: Contract verification should be available on testnets too.

```suggestion
## Verify your contract (mainnet deployment only)
```

File: content/en/smart-contracts/fundamentals/erc-20-quickstart/index.md [hunk 2]
```diff
@@ -213,6 +213,34 @@ Now that we've successfully compiled our contract, we need to deploy it somewher
...
+
+If you are deploying your contract to main net, we highly recommend verifying your contract, to upkeep security and transparency on the network.
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/erc-20-quickstart/index.md:218]:
```suggestion
If you are deploying your contract to mainnet, we highly recommend verifying your contract, to upkeep security and transparency on the network. 
```

File: content/en/smart-contracts/fundamentals/erc-20-quickstart/index.md [hunk 3]
```diff
@@ -213,6 +213,34 @@ Now that we've successfully compiled our contract, we need to deploy it somewher
...
 
+## Verify your contract (main net deployment only)
+
+If you are deploying your contract to main net, we highly recommend verifying your contract, to upkeep security and transparency on the network. 
+
+1. In Remix, within the **File Explorer** sidebar tab, under **contracts**, right click on your contract and choose "generate UML" to flatten the .sol file to ensure all components and tasks are included. 
+
+    <img width="779" alt="Screen Shot 2023-04-21 at 4 56 01 PM" src="https://user-images.githubusercontent.com/113331491/233750337-b15e51b3-aad5-4a50-bf62-6fc6625799d2.png">
+
+2. A contract ending in "_ flattened.sol" will be generated below your original contract. Ensure that the license and solidity version in the flattened contract is the same as in your original contract, edit as needed and click "save".
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/erc-20-quickstart/index.md:224]:
```suggestion
2. A contract ending in `_flattened.sol` will be generated below your original contract. Ensure that the license and Solidity version in the flattened contract is the same as in your original contract, edit as needed and click "save". 
```

Comment (raulk) [reply content/en/smart-contracts/fundamentals/erc-20-quickstart/index.md:224]: @trruckerfling what do you mean by "edit as needed"?

File: content/en/smart-contracts/fundamentals/erc-20-quickstart/index.md [hunk 4]
```diff
@@ -213,6 +213,34 @@ Now that we've successfully compiled our contract, we need to deploy it somewher
...
+
+2. A contract ending in "_ flattened.sol" will be generated below your original contract. Ensure that the license and solidity version in the flattened contract is the same as in your original contract, edit as needed and click "save". 
+
+    <img width="779" alt="Screen Shot 2023-04-21 at 4 59 01 PM" src="https://user-images.githubusercontent.com/113331491/233750517-5a95f65e-5cc2-46c4-8dc7-80ee248097de.png">
+
+3. Right click on the flattened contract and click "download" and save the file. Copy your deployed contract's address. 
+
+    <img width="779" alt="Screen Shot 2023-04-21 at 5 02 33 PM" src="https://user-images.githubusercontent.com/113331491/233750731-34bd6298-6e09-46be-a505-a8f4b90e69c8.png">
+
+4. Head over to a contract verifier of your choice. You can use [Filfox](https://filfox.info/en/contract) or [Beryx](https://beryx.zondax.ch/contract_verifier). We will use Filfox as an example here. Paste in your deployed contract's address, choose the same compiler version and license, as you used in Remix.
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/erc-20-quickstart/index.md:232]: Indicate where to find the compiler version under Remix.

File: content/en/smart-contracts/fundamentals/erc-20-quickstart/index.md (hunk 1 above)
Comment (trruckerfling) [reply content/en/smart-contracts/fundamentals/erc-20-quickstart/index.md:216]: is it important to verify contracts on testnet though?

File: content/en/smart-contracts/fundamentals/erc-20-quickstart/index.md (hunk 3 above)
Comment (trruckerfling) [reply content/en/smart-contracts/fundamentals/erc-20-quickstart/index.md:224]: 'edit as needed' referred to ensuring you paste the license and solidity version into the flattened contract. could remove the phrase and it wouldnt change the sentence, esp if it is confusing

File: content/en/smart-contracts/fundamentals/erc-20-quickstart/index.md (hunk 1 above)
Comment (raulk) [reply content/en/smart-contracts/fundamentals/erc-20-quickstart/index.md]: We should indicate it's an option too. Devs can test the verifiability early so they can be onto problems (theirs or the explorers's) to minimise surprise factors on mainnet.

**Comment by johnnymatthews** [comment]
Gonna set this to draft mode since it looks like you're still working on stuff @trruckerfling

File: content/en/smart-contracts/fundamentals/erc-20-quickstart/index.md [hunk 5]
```diff
@@ -213,6 +213,34 @@ Now that we've successfully compiled our contract, we need to deploy it somewher
 
 On the Filecoin network, a new set of blocks, also called a tipset, is created every thirty seconds. When deploying a contract, the transaction needs to be received by the network, and then the network needs to confirm the contract. This process takes around one to two tipsets to process -- or around 60 to 90 seconds.
 
+## Verify your contract (mainnet deployment only)
+
+If you are deploying your contract to mainnet, we highly recommend verifying your contract, to upkeep security and transparency on the network. 
+
+1. In Remix, within the **File Explorer** sidebar tab, under **contracts**, right click on your contract and choose "generate UML" to flatten the .sol file to ensure all components and tasks are included. 
+
+    <img width="779" alt="Screen Shot 2023-04-21 at 4 56 01 PM" src="https://user-images.githubusercontent.com/113331491/233750337-b15e51b3-aad5-4a50-bf62-6fc6625799d2.png">
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/erc-20-quickstart/index.md:222]: I think we want screenshots to live in this repo, right?

File: content/en/smart-contracts/fundamentals/erc-20-quickstart/index.md (hunk 3 above)
Comment (raulk) [reply content/en/smart-contracts/fundamentals/erc-20-quickstart/index.md]: @trruckerfling check out the suggestion -- the filename continues to be incorrect.

**Comment by ElPaisano** [comment]
> @trruckerfling we are making quick start way too big. I think `verify contracts` deserve a dedicated page for it, like:
> 
> * smart contract / Developing contracts / verify contracts

@longfeiWan9 @trruckerfling I've started work on this PR [Add contract verification guide to Developing Contracts](https://github.com/filecoin-project/filecoin-docs/pull/1993 ), which addresses this comment by creating a separate page. @johnnymatthews and I can reconcile these two PRs in the `staging` branch once these are both ready to go, and then get them merged into https://github.com/filecoin-project/filecoin-docs/tree/production

**Comment by johnnymatthews** [comment]
Closing this PR in favour of https://github.com/filecoin-project/filecoin-docs/pull/1993.

**Pull request closed without merging**